src/Nox.Integration/Services/NoxIntegration.cs

```diff
@@ -30,8 +30,10 @@ public NoxIntegration(ILogger logger, Solution.Integration definition)
         Description = definition.Description;
         MergeType = definition.MergeType;
         TransformType = definition.TransformationType;
-        if (definition.Source.Watermark == null) return;
-        var watermark = definition.Source.Watermark;
+
```

Comment:
Consider encapsulating this logic in a methods, since is not just assigning variables